src/IssueViz/TableControls/VSTableEventProcessorProvider.cs

```diff
@@ -0,0 +1,142 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using Microsoft.VisualStudio.Text.Classification;
+using Microsoft.VisualStudio.Utilities;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.IssueVisualization.TableControls
+{
+    [Export(typeof(ITableControlEventProcessorProvider))]
+    [Name(nameof(VSTableEventProcessorProvider))]
+    [Order(Before = Priority.Default)]
+    // We can specify multiple data source types and data source i.e. the same provider can be used for
+    // multiple table controls/sources
+    [DataSourceType(StandardTableDataSources.ErrorTableDataSource)]
+    [DataSource(SonarLintTableControlConstants.ErrorListDataSourceIdentifier)]
+    internal class VSTableEventProcessorProvider : ITableControlEventProcessorProvider
+    {
+        private readonly IIssueTablesSelectionMonitor issueTableSelectionMonitor;
+
+        [ImportingConstructor]
+        internal VSTableEventProcessorProvider(IIssueTablesSelectionMonitor issueTableSelectionMonitor)
+        {
+            this.issueTableSelectionMonitor = issueTableSelectionMonitor;
+        }
+
+        ITableControlEventProcessor ITableControlEventProcessorProvider.GetAssociatedEventProcessor(IWpfTableControl tableControl)
+        {
+            if (tableControl == null)
+            {
+                return null;
+            }
+
+            return new EventProcessor(tableControl, issueTableSelectionMonitor);
+        }
+
+        /// <summary>
+        /// Filters and processes events from a specific WPF table, raising the higher-level "selected issue changed"
+        /// changed when appropriate.
+        /// </summary>
+        internal /* for testing */ class EventProcessor : ITableControlEventProcessor
+        {
+            private readonly IWpfTableControl wpfTableControl;
+            private readonly IIssueTablesSelectionMonitor selectionMonitor;
+
+            public EventProcessor(IWpfTableControl wpfTableControl, IIssueTablesSelectionMonitor errorListSelectionMonitor)
+            {
+                this.wpfTableControl = wpfTableControl;
+                this.selectionMonitor = errorListSelectionMonitor;
+            }
+
+            void ITableControlEventProcessor.PostprocessSelectionChanged(TableSelectionChangedEventArgs e)
+            {
+                try
+                {
+                    IAnalysisIssue selectedIssue = null;
```

Comment:
Just for my understanding: this will be called on every selection change in the error list, right? So if we go from non-SL issue to non-SL issue, that will be called too?

Reply:
Yes, it will be called with a `null` issue. We want to know when a flow issue isn't highlighted so we can clear the flow pane and adornments.